recipes/podofo/all/conanfile.py

```diff
@@ -87,52 +84,58 @@ def requirements(self):
             self.requires("libunistring/0.9.10")
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd") and tools.Version(self.version) >= "0.9.7":
-            tools.check_min_cppstd(self, 11)
+        if self.info.settings.compiler.get_safe("cppstd") and Version(self.version) >= "0.9.7":
+            check_min_cppstd(self, 11)
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
-    @functools.lru_cache(1)
-    def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions["PODOFO_BUILD_LIB_ONLY"] = True
-        cmake.definitions["PODOFO_BUILD_SHARED"] = self.options.shared
-        cmake.definitions["PODOFO_BUILD_STATIC"] = not self.options.shared
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["PODOFO_BUILD_TOOLS"] = self.options.with_tools
+        tc.variables["PODOFO_BUILD_SHARED"] = self.options.shared
+        tc.variables["PODOFO_BUILD_STATIC"] = not self.options.shared
         if not self.options.threadsafe:
-            cmake.definitions["PODOFO_NO_MULTITHREAD"] = True
-        if not tools.valid_min_cppstd(self, 11) and tools.Version(self.version) >= "0.9.7":
-            cmake.definitions["CMAKE_CXX_STANDARD"] = 11
+            tc.variables["PODOFO_NO_MULTITHREAD"] = True
+        try:
+            check_min_cppstd(self, 11)
+        except ConanInvalidConfiguration:
+            if Version(self.version) >= "0.9.7":
+                tc.cache_variables["CMAKE_CXX_STANDARD"] = 11
```

Comment:
Why do you need C++11 if the upstream does not use it?

Reply:
I just copied the previous definition of the existing recipe